lib/getenv.c

```diff
@@ -27,25 +27,46 @@
 
 #include "memdebug.h"
 
-static
-char *GetEnv(const char *variable)
+static char *GetEnv(const char *variable)
 {
 #if defined(_WIN32_WCE) || defined(CURL_WINDOWS_APP)
   (void)variable;
   return NULL;
-#else
-#ifdef WIN32
-  char env[4096];
-  char *temp = getenv(variable);
-  env[0] = '\0';
-  if(temp != NULL)
-    ExpandEnvironmentStringsA(temp, env, sizeof(env));
-  return (env[0] != '\0')?strdup(env):NULL;
+#elif defined(WIN32)
+  /* This uses Windows API instead of C runtime getenv() to get the environment
+     variable since some changes aren't always visible to the latter. #4774 */
+  char *tmp;
+  DWORD rc;
+  char *buf = NULL;
+  DWORD bufsize = 0;
+
+  for(;;) {
+    /* It's possible for rc to be 0 if the variable was found but empty.
+       Since getenv doesn't make that distinction we ignore it as well. */
+    rc = GetEnvironmentVariableA(variable, buf, bufsize);
```

Comment:
I'm not 100% sure whether calling `GetEnvironmentVariable()` with a NULL buffer is fully supported. The [docs](https://docs.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-getenvironmentvariablea) don't appear to mention that use case. Maybe it would be safer to pass in a buffer of 1 byte, but still having bufsize==0?

Reply:
Ok I changed it to do that, would you check again

Reply:
This should work, but my thoughts have been more like:
````c
char _buf;
char *buf = &_buf;
DWORD bufsize = 0; // or perhaps init to 1?
...
rc = GetEnvironmentVariableA(variable, buf, bufsize);
````
and calling `realloc()` only after that (to avoid the first realloc() with size 1).